# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/wifi/nsWifiMonitor.h
# Commit: af2b512cb5ac
# Full Hash: af2b512cb5ac7a2b53224f5065f7d89fd8a503ed
# Author: David Parks <daparks@mozilla.com>
# Date: 2023-05-13 21:42:25
# Regressor Bug: 1810421
# File Overlap Count: 2
# Description:
#   Bug 1810421: Refactor wifi scanning to consolidate generic code r=necko-reviewers,kershaw,valentin a=reland CLOSED TREE
#   
#   Reorganizes the wifi-scanning code and makes more of it platform-generic
#   to ease the transition from polling the wifi to usually scanning only on
#   network changes.  This is essentially just moving files/code around and
# ==============================================================================

diff -r 3bb8a8b46f3c -r af2b512cb5ac netwerk/wifi/nsWifiMonitor.h
--- a/netwerk/wifi/nsWifiMonitor.h	Sat May 13 16:09:03 2023 +0000
+++ b/netwerk/wifi/nsWifiMonitor.h	Sat May 13 00:36:00 2023 +0000
@@ -18,10 +18,7 @@
 #include "nsIObserver.h"
 #include "nsTArray.h"
 #include "mozilla/Attributes.h"
-
-#ifdef XP_WIN
-#  include "win_wifiScanner.h"
-#endif
+#include "WifiScanner.h"
 
 extern mozilla::LazyLogModule gWifiMonitorLog;
 #define LOG(args) MOZ_LOG(gWifiMonitorLog, mozilla::LogLevel::Debug, args)
@@ -55,7 +52,8 @@
   NS_DECL_NSIRUNNABLE
   NS_DECL_NSIOBSERVER
 
-  nsWifiMonitor();
+  explicit nsWifiMonitor(
+      mozilla::UniquePtr<mozilla::WifiScanner>&& aScanner = nullptr);
 
  private:
   ~nsWifiMonitor() = default;
@@ -71,13 +69,12 @@
   mozilla::Atomic<bool> mThreadComplete;
   nsCOMPtr<nsIThread> mThread;  // only accessed on MainThread
 
+  // Background thread only (except in test).
+  mozilla::UniquePtr<mozilla::WifiScanner> mWifiScanner;
+
   nsTArray<nsWifiListener> mListeners MOZ_GUARDED_BY(mReentrantMonitor);
 
   mozilla::ReentrantMonitor mReentrantMonitor;
-
-#ifdef XP_WIN
-  mozilla::UniquePtr<WinWifiScanner> mWinWifiScanner;
-#endif
 };
 
 #endif